packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -0,0 +1,329 @@
+    actions?: Actions[];
+  };
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type Actions = {
+    type: string;
+    title: string;
+    value: string;
+  };
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type QueryMessageExtensionCard = {
+    contentType: string;
+    content: Record<string, unknown>; //object;
+    name?: string;
+    thumbnailUrl?: string;
+    contentUrl?: string;
+    fallbackHtml?: string;
+    signature?: string;
+  };
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type QueryMessageExtensionAttachment = QueryMessageExtensionCard & {
+    preview?: QueryMessageExtensionCard;
+  };
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type AttachmentLayout = 'list' | 'grid';
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type ComposeResultTypes = 'result' | 'auth' | 'config' | 'message' | 'silentAuth';
+  /*********** END RESPONSE TYPE ************/
+
+  /*********** BEGIN ERROR TYPE ***********/
+  export interface InvokeError {
+    errorCode: InvokeErrorCode;
+    message?: string;
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export enum InvokeErrorCode {
+    INTERNAL_ERROR = 'INTERNAL_ERROR', // Generic error
+  }
+  /*********** END ERROR TYPE ***********/
+
+  /**
+   * @beta
+   * @hidden
+   * Signals to the host to perform authentication using the given authentication parameters and then resend the request to the application specified by the app ID with the authentication result.
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId ID of the application backend to which the request and authentication response should be sent
+   * @param authenticateParameters Parameters for the authentication pop-up
+   * @param originalRequestInfo Information about the original request that should be resent
+   * @returns A promise that resolves to the IInvokeResponse from the application backend and rejects with InvokeError if the host encounters an error while authenticating or resending the request
+   */
+  export function authenticateAndResendRequest(
+    appId: string,
+    authenticateParameters: authentication.AuthenticatePopUpParameters,
+    originalRequestInfo: OriginalRequestInfo,
+  ): Promise<IInvokeResponse> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    // Ask the parent window to open an authentication window with the parameters provided by the caller.
+    return sendMessageToParentAsync<[InvokeError, IInvokeResponse]>(
+      'externalAppAuthentication.authenticateAndResendRequest',
+      [
+        appId,
+        originalRequestInfo,
```

Comment:
does the order matters here? this is appId, originalRequestInfo, and then Authenticate parameters but the doc shows appId, authenticate parameters and then originalRequestInfo at last.

Reply:
It just has to match the order in the host SDK, I'll double check that it does

Reply:
The doc is wrong, I'll update it.

Reply:
Actually, the order of parameters, while important, isn't reflected in the doc, it's purely an implementation detail. In any case, they appear to match currently.